src/proto/blocks.proto

```diff
@@ -79,6 +79,19 @@ message FullProposal {
   }
 }
 
+message DecidedValue {
+  oneof value {
+    Block block = 2;
+    ShardChunk shard = 3;
+  }
+}
+
+message ReadNodeMessage {
+  oneof read_node_message {
```

Comment:
Why do we need a oneof here?

Reply:
This is in case we add other messages specific to read nodes. We can add something without a breaking protocol change.